### PR TITLE
Merge CAN I/O implementation

### LINKED ISSUE #22: Get main branch to a working state
Currently the working code is all located on `hi-can-dev` and isn't in a merge-able state. To get it ready to merge, it needs to have tests written (probably closing #10), all placeholders removed, and all template/example code replaced.

### PR DESCRIPTION
**Pull request by RandomSpaceship**
PR to track progress on merging hi-can-dev into main. Will close #22.

### CODE REVIEW

**Review by DingoOz — APPROVED** [review]
Represents current rules.

**Review by DingoOz — APPROVED** [review]
This link is correct and useful.

**Comment by RandomSpaceship** [comment]
Closes #28.

File: software/ros_ws/src/perseus_hardware/src/mcb_system.cpp [hunk 1]
```diff
@@ -1,70 +1,292 @@
...
-hardware_interface::CallbackReturn McbSystemHardware::on_init(const hardware_interface::HardwareInfo& info)
+CallbackReturn McbSystemHardware::on_init(const HardwareInfo& info)
 {
-    if (const auto& ret = hardware_interface::SystemInterface::on_init(info); ret != hardware_interface::CallbackReturn::SUCCESS)
+    if (const auto& ret = SystemInterface::on_init(info);
+        ret != CallbackReturn::SUCCESS)
         return ret;
 
-    return hardware_interface::CallbackReturn::SUCCESS;
+    CHECK_HARDWARE_PARAMETER_EXISTS(get_logger(), info, "can_bus");
```
Comment (DingoOz) [inline software/ros_ws/src/perseus_hardware/src/mcb_system.cpp:32]: The call to get_logger is causing build errors when I try to build this. Still happy for PR to be accepted as it appears that the errors are contained to get_logger().

Build error is:
```
/home/dingo/perseus-v2/software/ros_ws/src/perseus_hardware/src/mcb_system.cpp: In member function ‘virtual hardware_interface::CallbackReturn perseus_hardware::McbSystemHardware::on_init(const hardware_interface::HardwareInfo&)’:
/home/dingo/perseus-v2/software/ros_ws/src/perseus_hardware/src/mcb_system.cpp:32:37: error: ‘get_logger’ was not declared in this scope; did you mean ‘rclcpp::get_logger’?
   32 |     CHECK_HARDWARE_PARAMETER_EXISTS(get_logger(), info, "can_bus");
```

**Review by DingoOz — APPROVED** [review]

**Comment by DingoOz** [comment]
PR looks good to go. Well done on a huge effort. I think you've got 100% of the humble references. The changes to xbox_controller make sense to me.   hi-can all looks good but I can't comment too much but I can see that it builds and we used it on Perseus actual, so I assume it is good :)  Using my limited nix knowledge, it all looks sensible. All the doc links and info appear to function correctly. 

The only issue was the get_logger in mcb, but see comment above.

Thanks.

**Pull request merged**